packages/terra-navigation-prompt/docs/README.md

```diff
@@ -0,0 +1,138 @@
+# Terra Navigation Prompt
+
+The NavigationPrompt and NavigationPromptCheckpoint components allow for the detection and management of components with unsaved state.
+
+## Getting Started
+
+- Install with [npmjs](https://www.npmjs.com):
+  - `npm install terra-navigation-prompt`
+- Import the module
+  - `import NavigationPrompt, { NavigationPromptCheckpoint } from 'terra-navigation-prompt';`
+
+## Rationale
+
+A React component has complete control over its own rendering logic. It can render child components and unmount those child components as it sees fit. It follows that any individual component has no direct control over its own mounting and unmounting; it is rendered (or not) at the discretion of its parent. A component must rely on its lifecycle events to determine how it is being implemented.
+
+However, a component's children can be stateful and responsible for important user input or interactions. If the user can navigate away from that child component, it may be necessary for that child component to notify the user that their workflow is incomplete or that some transient state has not yet been persisted. If the component has no control over its own unmounting, how can it communicate this state to the overall application before the component is already being unmounted?
+
+The most straightforward solution is to lift the state up. If the state is lifted into a parent component, it can be more directly managed by the component responsible for navigation, or it could be persisted indefinitely. **This is the strategy that should be implemented by most components.** However, for very large applications or highly composable applications built out of many stateful components, lifting the state may not be feasible.
```

Comment:
The most straightforward solution is to lift the state up. If the **child component's** state is lifted into a parent component, it can **either** be ~more~ directly managed by the **parent** component **that is** responsible for navigation, or it **can** ~could~ be persisted indefinitely.